### DISCUSSION TITLE
Why are there 2 versions of my statically rendered page?

### DISCUSSION THREAD

**Discussion by codinginflow** [vercel/next.js#50114] [App Router]
For testing purposes, I created a page where I fetch an image from an **endpoint that returns random data** on each request. Since my page doesn't use any **dynamic features**, we only make the fetch request at **build time** and therefore get the same image on every page request.

This seems to work, except that I get **2 different pages** instead of one. When I navigate via **Links**, a different image is shown than after **refreshing the page**. Subsequent refreshes keep showing the same image tho, so static rendering is definitely working (I can also see it in the build console).

![app dir static page bug](https://github.com/vercel/next.js/assets/52977034/7b265984-0a7e-4fd8-bd31-7996c909ed4e)

Here is the code for that page:

```
import Image from "next/image";
import { UnsplashImage } from "../../../models/unsplash-image";
import { Alert } from "@/components/bootstrap";
import Link from "next/link";

export const metadata = {
    title: "Static Fetching - NextJS 13.4 Image Gallery",
}

export default async function Page() {

    const response = await fetch("https://api.unsplash.com/photos/random?client_id=" + process.env.UNSPLASH_ACCESS_KEY);

    const image: UnsplashImage = await response.json();

    const width = Math.min(500, image.width);
    const height = (width / image.width) * image.height;

    return (
        <div className="d-flex flex-column align-items-center">
            <Alert>
                This page <strong>fetches and caches data at build time</strong>.
                Even though the Unsplash API always returns a new image, we see the same image after refreshing the page until we compile the project again.
            </Alert>

            <Image
                src={image.urls.raw}
                alt={image.description}
                width={width}
                height={height}
                className="rounded shadow mw-100 h-100"
            />
            <p>by <Link href={"/users/" + image.user.username}>{image.user.username}</Link></p>
        </div>
    );
}
```

**Comment by icyJoseph** [comment]
I tried to reproduce this, but I always get the same image. What Next.js version have you got? I tried a few times and nothing, maybe I should add delays?

https://github.com/vercel/next.js/assets/21013447/af091ee4-7b36-4c9d-bc36-b6b62f714a36

I was thinking that maybe there's two renderings going on, I've heard about this. Like, in one Next.js is pre-rendering the HTML, and in the other, the RSC wire-format/JSON data that drives the page.

```jsx
import Image from "next/image";
import Link from "next/link";

export const metadata = {
  title: "Static Fetching - NextJS 13.4 Image Gallery",
};

const randomImage = () => {
  const seed = 1 + Math.floor(100 * Math.random());
  console.log("using seed", seed);
  return `https://picsum.photos/seed/${seed}/300/300`;
};

export default async function Page() {
  const url = randomImage();

  return (
    <div className="d-flex flex-column align-items-center">
      <Image
        src={url}
        alt="random img from picsum"
        width={200}
        height={300}
        className="rounded shadow mw-100 h-100"
      />
      <p>by Picsum</p>

      <div>
        Back home <Link href="/">home</Link>
      </div>
    </div>
  );
}

```

Something like that one, w/ next 13.4.3, I see that it runs twice. The first time it seems, it generates the HTML for the page, the second time, it generates the RSC data wire/json thing.

https://github.com/vercel/next.js/assets/21013447/04e9c888-4c30-4bad-a51c-2e77972d822e

Odd that it needs to do the run twice though, I don't think GSP did that, pretty sure it reused the same GSP call to generate HTML and JSON for the page. 

Hard to argue for render purity here too, since it is an async call... 🤔

**Reply by codinginflow** [reply]
Thank you for trying it out!

I'm using `next 13.4.3`

Could the different page be caused because I have a `Link` to that static page in the nav bar and NextJS somehow caches the wrong page in the router?

**Reply by codinginflow** [reply]
@icyJoseph I think the bug only happens if there is another page that fetches data from the same endpoint. Are the caches interfering with each other? Does every page get its own cache even if they fetch from exact same API endpoint?
